src/modm/platform/adc/stm32f3/adc_impl.hpp.in

```diff
@@ -196,17 +245,74 @@ modm::platform::Adc{{ id }}::startConversion(void)
 }
 
 bool
-modm::platform::Adc{{ id }}::isConversionFinished(void)
+modm::platform::Adc{{ id }}::isConversionFinished()
 {
 	return static_cast<bool>(getInterruptFlags() & InterruptFlag::EndOfRegularConversion);
 }
 
+void
+modm::platform::Adc{{ id }}::startInjectedConversionSequence()
+{
+	acknowledgeInterruptFlags(InterruptFlag::EndOfInjectedConversion |
+			InterruptFlag::EndOfInjectedSequenceOfConversions);
+
+	ADC{{ id }}->CR |= ADC_CR_JADSTART;
+}
+
+bool
+modm::platform::Adc{{ id }}::setInjectedConversionChannel(uint_fast8_t index, Channel channel,
+														  SampleTime sampleTime)
+{
+	if (index >= 4) {
+		return false;
+	}
+	if (!configureChannel(channel, sampleTime)) {
+		return false;
+	}
+
+	static_assert(ADC_JSQR_JSQ2_Pos == ADC_JSQR_JSQ1_Pos + 6);
+	static_assert(ADC_JSQR_JSQ3_Pos == ADC_JSQR_JSQ2_Pos + 6);
+	static_assert(ADC_JSQR_JSQ4_Pos == ADC_JSQR_JSQ3_Pos + 6);
+
+	const uint32_t pos = ADC_JSQR_JSQ1_Pos + 6 * index;
+	const uint32_t mask = ADC_JSQR_JSQ1_Msk << (6 * index);
+	ADC{{ id }}->JSQR = (ADC{{ id }}->JSQR & ~mask) | (static_cast<uint32_t>(channel) << pos);
+	return true;
+}
+
+template<class Gpio>
+bool
+modm::platform::Adc{{ id }}::setInjectedConversionChannel(uint_fast8_t index,
+														  SampleTime sampleTime)
+{
+	return setInjectedConversionChannel(index, getPinChannel<Gpio>(), sampleTime);
+}
+
+bool
+modm::platform::Adc{{ id }}::setInjectedConversionSequenceLength(uint_fast8_t length)
+{
+	if (length < 1 || length > 4) {
```

Comment:
```suggestion
	if (length < 1 or length > 4) {
```

Reply:
If you strongly prefer `or` over `||` I can change it, but `||` and `&&` are generally the more standard and commonly used way to write those operators in C and C++ (and they are also written like that in Rust, Java, Javascript, C#, PHP, Swift, Kotlin, Scala, ...). I'm so used to seeing those that `or` doesn't make it more readable for me at least, but more foreign for C++ code. In C the alternative tokens like `and` and `or` are even macros defined in `iso646.h`.
modm has both variants and isn't really consistent on that matter, hm ...

Reply:
The rationale was that `|` and `||` are easy to confuse visually compared to `|` and `or`. It differentiates bit from boolean operations more easily.

We need to figure out a non-annoying way to actually use 
 clang-format/clang-tidy so we don't manually have to format such things in the future.

Reply:
Also in Python || and && doesn't exist and I've written way too much Python in recent years ;-)

I see you conveniently left that one out of your list :-P

Reply:
Also we use `bool`, `true` and `false` and not `int`, `1` and `0`, so we should use `or`, `and` etc. The common way is wrong, WRONG I TELL YA!1!!

Reply:
It's not really important what we choose here, as long as we choose only one option and apply it consistently. If you strongly prefer `and` and `or`, we can have them. 

> Also we use bool, true and false and not int, 1 and 0, so we should use or, and etc. The common way is wrong, WRONG I TELL YA!1!!

😱, using `int`, `1` and `0` for `bool`s in C++? If that's the common advice you get, I'd consider looking for a different source :-P.

Reply:
Yes, use `or`please.